### ISSUE TITLE
Google Accounts integration signs out Gmail when different account is authenticated

### ISSUE THREAD

**Issue by mattbasta** [mozilla/persona#3714]
If I'm signed into Gmail with bastawhiz@gmail.com (my primary email) and I want to sign in as mattbasta@gmail.com. I get the message:

```
You're currently signed into Google as bastawhiz@gmail.com.
If you want to use mattbasta@gmail.com, log out of Google and sign in again.
```

This is infinitely frustrating. Forcing a Google sign out logs out my gmail, google calendar, google drive, google voice (!), any services which use Google Accounts to manage sessions, etc.

Is there no way to authenticate with Google without killing all my existing sessions across all my open tabs?

**Comment by seanmonstar** [comment]
You can be signed in as both. In Gmail, go to the account button, and add account. Then, when you use Persona, and we forward to Gmail, it will show a picker.

**Comment by mattbasta** [comment]
For anyone that's not signed in to both simultaneously (a great example is YouTube users that had their accounts upgraded), that isn't a real option. It's also not a great option if your Google Account doesn't support the service that you're using (i.e.: you've got a Google enterprise account but it doesn't have Gmail access) simply because it takes an act of god to get your account picker set up properly.

**Comment by seanmonstar** [comment]
How so? For example, we have Google Apps for Your Domain accounts with Mozilla. But those accounts only have Google Drive access. I tested this with my gmail account, and my mozilla account. You can have both signed in, even if a certain property isn't available to one of the accounts.

Then, when you're passed to the OpenID endpoint, it will show all your accounts in the picker.

**Comment by mattbasta** [comment]
With my personal Google Apps account, I can never get it to stay in the
account picker. It might also have to do with having set up account
forwarding for Gmail, not sure. Either way, this is still exceptionally
frustrating for users that haven't set up multiple accounts.

**Comment by zaach** [comment]
@ryanfeeley Should the copy mention multiple sign-in as an alternative to logging out/in?

**Comment by ryanfeeley** [comment]
You mean highlight @seanmonstar's tip about adding one account to another?

**Comment by zaach** [comment]
Yep, perhaps with a [link](https://support.google.com/mail/answer/1721977?hl=en) for explanation.

**Comment by krupa** [comment]
I too find this infinitely annoying. I maintain separate gmail accounts for my personal email and bugmail (for the sake of my sanity) and having to sign out of my personal email just to be able to log in to marketplace is not optimal.

**Comment by callahad** [comment]
This is working as designed. We suggest using the multiple sign-in feature at Google if you need this. See Zaach's link for more info.

**Comment by mattbasta** [comment]
You're missing the point and I'd like to ask you to reopen this issue. It's not that Persona is behaving not-as-intended, it's that the intended behavior is extremely frustrating.

`mattbasta@gmail.com` is a forwarding address, and I've used it with Persona forever. There's no reason I'd ever want to be signed into GA with it, but now Persona is forcing me to set up multiple accounts (which are only good for the current session, I might add) so that I can log into Persona-enabled sites without being forcibly signed out of the many Google services that I use. If I want to log into a Persona-enabled site with yet another gmail address that I don't ever use, I either have to think ahead and set up multiple accounts for it or be forcibly signed out of the other two accounts that I've already set up so that I can work with Persona.

When I sign in as `basta@mozilla.com`, I'm not logged out of `intranet.m.o` or Zimbra because I'm signed in as `mbasta@mozilla.com` on those services. Clearly this problem isn't technically insurmountable.

If at least two people inside Mozilla are experiencing this (I can name at least two others that simply haven't commented on this bug), I can only imagine that it's going to be a PITA for many more non-Mozillians.

Perhaps the solution here is to work with Google to get a version of the OpenID endpoint that isn't tied to the user's current session.
